cluster_pack/spark/spark_config_builder.py

```diff
@@ -20,20 +23,52 @@ def add_packaged_environment(ssb: SparkSession.Builder, archive: str):
         os.environ['PYSPARK_PYTHON'] = "./condaenv/bin/python"
         os.environ['PYSPARK_DRIVER_PYTHON'] = 'python'
 
+    if not _get_value(ssb, "spark.master") == "yarn":
+        _logger.info("Not running on yarn. Adding archive to spark.files")
+        _add_or_merge(ssb, "spark.files", f"{archive}")
+
 
 def add_editable_requirements(ssb: SparkSession.Builder):
     for requirement_dir in packaging.get_editable_requirements().values():
         py_archive = packaging.zip_path(requirement_dir)
         _add_archive(ssb, py_archive)
 
 
+def add_s3_params(ssb: SparkSession.Builder,  fs_args: Dict[str, Any] = {}):
+    ssb.config("spark.hadoop.fs.s3a.impl", "org.apache.hadoop.fs.s3a.S3AFileSystem")
+    ssb.config("spark.hadoop.fs.s3a.path.style.access", "true")
+    if "key" in fs_args:
+        ssb.config("spark.hadoop.fs.s3a.access.key", fs_args["key"])
+    if "secret" in fs_args:
+        ssb.config("spark.hadoop.fs.s3a.secret.key", fs_args["secret"])

```

Comment:
I would have named the args 'access_key' and 'secret_key'

Reply:
I reused the args of:
https://s3fs.readthedocs.io/en/latest/api.html#s3fs.core.S3FileSystem

such that we can just do:
```
s3fs = new S3FileSystem(**fs_args)
```

Reply:
also applies to this (client_kwargs : dict of parameters for the botocore client)
```
 if "client_kwargs" in fs_args:
        ssb.config("spark.hadoop.fs.s3a.endpoint", fs_args["client_kwargs"]["endpoint_url"])
```
we could directly pass the url also

Reply:
Personally I prefer just reusing the args from S3FileSystem but it I agree it is debatable. What happens if we want to use the new S3 client of pyarrow (with other names for the args, but it seems not mature yet)

Reply:
Ok it's fine for me, I didn't check S3FileSystem documentation before my comment